examples/vision/deeplabv3_plus.py

```diff
@@ -0,0 +1,298 @@
+        dilation_rate=1,
+        padding="same",
+        use_bias=use_bias,
+        kernel_initializer=tf.keras.initializers.HeNormal(),
+    )(block_input)
+    x = tf.keras.layers.BatchNormalization()(x)
+    return tf.nn.relu(x)
+
+
+def AtrousSpatialPyramidPooling(aspp_input):
+
+    dims = tf.keras.backend.int_shape(aspp_input)
+
+    layer = tf.keras.layers.AveragePooling2D(pool_size=(dims[-3], dims[-2]))(aspp_input)
+    layer = convolution_block(layer, kernel_size=1, use_bias=True)
+
+    out_pool = tf.keras.layers.UpSampling2D(
+        size=(dims[-3] // layer.shape[1], dims[-2] // layer.shape[2]),
+        interpolation="bilinear",
+    )(layer)
+
+    out_1 = convolution_block(aspp_input, kernel_size=1, dilation_rate=1)
+    out_6 = convolution_block(aspp_input, kernel_size=3, dilation_rate=6)
+    out_12 = convolution_block(aspp_input, kernel_size=3, dilation_rate=12)
+    out_18 = convolution_block(aspp_input, kernel_size=3, dilation_rate=18)
+
+    layer = tf.keras.layers.Concatenate(axis=-1)(
+        [out_pool, out_1, out_6, out_12, out_18]
+    )
+
+    output = convolution_block(layer, kernel_size=1)
+
+    return output
+
+
+"""
+The encoder features are first bilinearly upsampled by a factor of 4 and then 
+concatenated with the corresponding low-level features from the network backbone that
+have the same spatial resolution. For this example, we would be 
+using Resnet101 pre-trained on ImageNet as the backbone model and we would use 
+the low-level features from the Conv2 block of the backbone.
+"""
+
+
+def DeeplabV3Plus():
+    model_input = tf.keras.Input(shape=(IMAGE_SIZE, IMAGE_SIZE, 3))
+    resnet101 = tf.keras.applications.ResNet101(
+        weights='imagenet', include_top=False, input_tensor=model_input)
+    layer = resnet101.get_layer('conv4_block6_2_relu').output
+    layer = AtrousSpatialPyramidPooling(layer)
+    
+    input_a = tf.keras.layers.UpSampling2D(
+        size=(
+            IMAGE_SIZE // 4 // layer.shape[1],
+            IMAGE_SIZE // 4 // layer.shape[2]
+        ), interpolation = 'bilinear'
+    )(layer)
+    
+    input_b = resnet101.get_layer('conv2_block3_2_relu').output
+    input_b = convolution_block(input_b, n_filters=48, kernel_size=1)
+    
+    layer = tf.keras.layers.Concatenate(axis=-1)([input_a, input_b])
+    layer = convolution_block(layer)
+    layer = convolution_block(layer)
+    layer = tf.keras.layers.UpSampling2D(
+        size=(
+            IMAGE_SIZE // layer.shape[1],
+            IMAGE_SIZE // layer.shape[2]
+        ), interpolation='bilinear')(layer)
+    
+    model_output = tf.keras.layers.Conv2D(
+        NUM_CLASSES, kernel_size=(1, 1), padding='same')(layer)
+    
+    return tf.keras.Model(inputs=model_input, outputs=model_output)
+
+"""
+## Training
+"""
+
+"""
+We would train the model using Sparse Categorical Cross-entropy as the loss function and
+use Adam as the optimizer.
+"""
+
+loss = tf.losses.SparseCategoricalCrossentropy(from_logits=True)
```

Comment:
`keras.losses. SparseCategoricalCrossentropy`

Reply:
Added the following imports at the top and adjusted the layer, model, loss, and optimizer instantiations

```python3
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
```